src/components/FilterGroup.tsx

```diff
@@ -85,17 +96,69 @@ export function FilterGroup({
       {renderTitle()}
       <CollapsibleSection className={cssClasses.optionsContainer}>
         {searchable && <SearchInput className={cssClasses.searchInput} />}
-        {filterOptions.map(o => {
-          return (
-            <CheckboxOption
-              {...o}
-              key={o.displayName || o.value.toString()}
-              customCssClasses={cssClasses}
-            />
-          );
-        })}
+        <CheckboxOptions
+          filterOptions={filterOptions}
+          showMoreLimit={showMoreLimit}
+          cssClasses={cssClasses}
+          showAll={showAll}
+          setIsLimited={setIsLimited} />
+        {isLimited &&
+          /* eslint-disable-next-line react-perf/jsx-no-new-function-as-prop */
+          <button className='text-primary py-1 text-sm' onClick={() => setShowAll(!showAll)}>
+            {showAll ? 'Show Less' : 'Show More'}
+          </button>
+        }
         {children}
       </CollapsibleSection>
     </FilterGroupProvider>
   );
 }
+
+function CheckboxOptions({
+  filterOptions,
+  showMoreLimit,
+  cssClasses,
+  showAll,
+  setIsLimited
+}: {
+  filterOptions: FilterOptionConfig[],
+  showMoreLimit: number,
+  cssClasses: CheckboxCssClasses,
+  showAll: boolean,
+  setIsLimited: (limited: boolean) => void
+}) {
+  const searchUtilities = useSearchUtilities();
+  const { searchValue } = useFilterGroupContext();
+
+  const shouldRenderOption = (
+    option: FilterOptionConfig
+  ) => {
+    if (isUndefined(option.displayName)) {
+      option.displayName = option.value.toString();
+    }
+    return searchUtilities.isCloseMatch(option.displayName, searchValue);
+  };
+
+  let displayedOptions = filterOptions.filter(shouldRenderOption).map(o => {
+    return (
+      <CheckboxOption
+        {...o}
+        key={o.displayName}
+        customCssClasses={cssClasses}
+      />
+    );
+  });
+
+  if (displayedOptions.length > showMoreLimit) {
+    displayedOptions = displayedOptions.slice(0, showAll ? filterOptions.length : showMoreLimit);
```

Comment:
nit: it would be more accurate for the ternary operator to resolve to `displayedOptions.length` rather than `filterOptions.length` when `showAll` is true